src/Appwrite/Migration/Version/V11.php

```diff
@@ -0,0 +1,683 @@
+<?php
+
+namespace Appwrite\Migration\Version;
+
+use Appwrite\Database\Database as OldDatabase;
+use Appwrite\Database\Document as OldDocument;
+use Appwrite\Migration\Migration;
+use Exception;
+use PDO;
+use Redis;
+use Swoole\Runtime;
+use Throwable;
+use Utopia\Abuse\Adapters\TimeLimit;
+use Utopia\Audit\Audit;
+use Utopia\Cache\Cache;
+use Utopia\CLI\Console;
+use Utopia\Cache\Adapter\Redis as RedisCache;
+use Utopia\Config\Config;
+use Utopia\Database\Adapter\MariaDB;
+use Utopia\Database\Database;
+use Utopia\Database\Document;
+use Utopia\Database\Exception\Limit;
+use Utopia\Database\Exception\Authorization as ExceptionAuthorization;
+use Utopia\Database\Exception\Structure;
+use Utopia\Database\Query;
+use Utopia\Database\Validator\Authorization;
+
+global $register;
+
+class V11 extends Migration
+{
+    protected Database $dbInternal;
+    protected Database $dbExternal;
+    protected Database $dbConsole;
+
+    protected array $oldCollections;
+    protected array $newCollections;
+
+    public function __construct(PDO $db, Redis $cache = null, array $options = [])
+    {
+        parent::__construct($db, $cache, $options);
+        $this->options = array_map(fn ($option) => $option === 'yes' ? true : false, $this->options);
+
+        if (!is_null($cache)) {
+            $cacheAdapter = new Cache(new RedisCache($this->cache));
+            $this->dbInternal = new Database(new MariaDB($this->db), $cacheAdapter);
+            $this->dbExternal = new Database(new MariaDB($this->db), $cacheAdapter);
```

Comment:
I had to go looking to see if `$project` was in scope, so I think a comment helps here.
```suggestion
            $this->dbInternal = new Database(new MariaDB($this->db), $cacheAdapter); // namespace is set on execution
            $this->dbExternal = new Database(new MariaDB($this->db), $cacheAdapter); // namespace is set on execution
```